compiler/src/dotty/tools/dotc/transform/LambdaLift.scala

```diff
@@ -338,11 +338,17 @@ object LambdaLift {
             else (topClass, JavaStatic)
           }
           else (lOwner, EmptyFlags)
+        // Drop Module because class is no longer a singleton in the lifted context.
+        val preInitFlags = local.flags &~ Module | Private | Lifted | maybeStatic
+        // Drop Final when a method is lifted into a trait.
+        // According to the JVM specification, a method declared inside interface cannot have the final flag.
+        // "Methods of interfaces may have any of the flags in Table 4.6-A set except ACC_PROTECTED, ACC_FINAL, ..."
+        // (https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.6)
+        val initFlags = if (!local.isClass && (newOwner.flags is Trait)) preInitFlags &~ Final else preInitFlags
```

Comment:
Why don't you test `local.is(Method)` instead of `!local.isClass`?

Reply:
Personal test: I think it is clearer to use a `var` rather than coming up with new unique names. Ie.
```scala
var initFlags = ...
if (...) initFlags = initFlags &~ Final
```

Reply:
I think your suggestion is better than my original code. I'll fix it. Thank you!